modules/top-level/output.nix

```
@@ -339,7 +339,9 @@ in
           initSource;
 
       extraWrapperArgs = builtins.concatStringsSep " " (
-        (optional (
+        # Setting environment variables in the wrapper
+        (lib.mapAttrsToList (var: value: "--set ${var} ${lib.escapeShellArg value}") config.env)
```

or should this be `escapeShellArgs` for the entire list? not too familiar with its usage

---

I'd personally prefer this:

```suggestion
        (lib.mapAttrsToList (name: value: lib.escapeShellArgs [ "--set" name value ]) config.env)
```